### ISSUE TITLE
ChangeLog formatting

### ISSUE THREAD

**Issue by thekid** [xp-framework/rfc#273]
# Scope of Change

ChangeLog formatting will be changed to use markdown syntax, and the filename will be changed to `ChangeLog.md`.
# Rationale

Easier viewing on GitHub.
# Functionality
### Basic format

The basic ChangeLog uses second level headers for releases. The newest release has a placeholder. Release candidates, alpha and beta versions are usually shown explicitely. Every line is followed by `@handle` referring to the author(s) of this change.

``` markdown
## ?.?.? / ????-??-??

* Deleted ... (@handle)

## [Version] / [YYYY-MM-DD]

* Changed ... (@handle)
* Fixed ... (@handle)
* Added ... (@handle, co-author)

## [Version] / [YYYY-MM-DD]

* Removed ... (@handle)
```
### Release notes

If a release has a very special meaning, release notes may be added _beore_ the bullet-pointed list of changes:

``` markdown
## 5.9.0 / 2013-05-05
After roughly a year of development, this release forms the first release in the 5.9-SERIES. 
The minimum runtime PHP version required is now PHP 5.3.0. Be sure to read the 
[migrating to 5.9](https://...) document before upgrading!

* Removed ... (@handle)
```
### For larger ChangeLogs

Optionally, the changes may be divied into sections:

``` markdown
### Heads up!

* Changed ... (@handle)
* Deprecated ... (@handle)

### RFCs

* Implemented RFC [XXXX]: [RFC Title] (@handle)

### Bugfxies

* Fixed issue #[XXXX]: [Issue title] (@handle)

### Features

* Implemented pull request #[XXXX]: [Pull request title] (@handle)
```

RFC here refers to XP Technology RFCs, https://github.com/xp-framework/rfc
# Security considerations
# Speed impact
# Dependencies
# Related documents

Examples:
- https://github.com/bower/bower/blob/master/CHANGELOG.md
- https://github.com/chaplinjs/chaplin/blob/master/CHANGELOG.md
- http://rubydoc.info/gems/rspec-core/file/Changelog.md

**Comment by kiesel** [comment]
Like this.

**Comment by thekid** [comment]
https://github.com/xp-framework/xp-framework/blob/master/core/ChangeLog.md

**Comment by thekid** [comment]
https://github.com/xp-framework/xp-language/blob/master/compiler/ChangeLog.md

**Comment by thekid** [comment]
The ChangeLogs for the respective version are now delivered alongside with the archive files for modules build with the [XP Build system](https://github.com/xp-framework/build):

``` sh
$ xpi add xp-framework/xp-language@1.10.2
Module<xp-framework/xp-language> -> io.Folder(uri= ...\.xp\xp-framework\xp-language@1.10.2\)
Release 1.10.2 published 2013-05-20T16:25:11+02:00
>> xp-compiler-1.10.2.xar
>> xp-compiler-test-1.10.2.xar
>> ChangeLog
# ^^^^^^^^^^
#   Here it is!
```

The `ChangeLog` file contains only the part of the changelog relevant to the respective release:

``` sh
$ cat xp-framework/xp-language@1.10.2/ChangeLog
* Implemented RFC 273: ChangeLog formatting - (@thekid)
* Fixed issue #32 - Fatal error during compilation (@thekid)
```
